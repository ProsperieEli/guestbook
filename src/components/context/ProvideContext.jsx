import React from 'react';
import { useState, createContext } from 'react';

//create context is a function.
//create context is a function that can pass global state/data.
export const authContext = createContext();

export default function ProvideContext({ children }) {
  const [users, setUsers] = useState(null);

  const login = (userName, password) => {
    const loginSuccess =
      userName === process.env.AUTH_USERNAME &&
      password === process.env.AUTH_PASSWORD;

    if (loginSuccess) setUsers({ userName });
    return loginSuccess;
  };

  const logout = (callback) => {
    setUsers(null);
    callback();
  };

  return (
    //this tells it what to pass down to anything that is wrapped inside it.
    <authContext.Provider value={{ users, login, logout }}>
      {children}
    </authContext.Provider>
  );
}
