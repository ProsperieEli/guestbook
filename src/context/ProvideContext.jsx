import React from 'react';
import { useState, createContext } from 'react';

//create context is a function.
//create context is a function that can pass global state/data.
export const authContext = createContext();

export default function ProvideContext({ children }) {
  const [user, setUser] = useState(null);

  const login = (username, password) => {
    const loginSuccess =
      username === process.env.AUTH_USERNAME &&
      password === process.env.AUTH_PASSWORD;

    if (loginSuccess) {
      setUser(username);
      console.log('hi', username);
    }
    return loginSuccess;
  };

  const logout = () => {
    setUser(null);
  };

  return (
    //this tells it what to pass down to anything that is wrapped inside it.
    <authContext.Provider value={{ user, login, logout }}>
      {children}
    </authContext.Provider>
  );
}
