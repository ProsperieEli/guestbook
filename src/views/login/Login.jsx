import React from 'react';
import { useAuth } from '../../hooks/useAuth';
import { useHistory, useLocation } from 'react-router-dom';
import { useState } from 'react/cjs/react.development';

export default function Login() {
  const history = useHistory();
  const location = useLocation();
  const auth = useAuth();
  const [username, setUsername] = useState('');
  const [password, setPassword] = useState('');

  const [error, setError] = useState(null);

  let { from } = location.state || { from: { pathname: '/' } };

  const handleLogin = (e) => {
    e.preventDefault();
    const loggedin = auth.login(username, password);

    loggedin ? history.replace(from) : setError('Login failed');
  };

  return (
    <>
      <h1>Welcome! Login here!</h1>
      <form onSubmit={handleLogin}>
        <label htmlFor="userName">UserName</label>
        <input
          id="userName"
          name="userName"
          type="text"
          value={username}
          onChange={(event) => setUsername(event.target.value)}
        />
        <label htmlFor="password">Password</label>
        <input
          id="password"
          name="password"
          type="password"
          value={password}
          onChange={(event) => setPassword(event.target.value)}
        />
        <button>Login</button>
      </form>
      <h3>{error}</h3>
    </>
  );
}
