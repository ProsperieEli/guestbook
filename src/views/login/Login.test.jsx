import React from 'react';
import { render } from '@testing-library/react';
import ProvideContext from '../../context/ProvideContext';
import Login from './Login';
import { MemoryRouter } from 'react-router-dom';

it('Should login', () => {
  const { container } = render(
    <MemoryRouter initialEntries={['/login']}>
      <ProvideContext>
        <Login />
      </ProvideContext>
    </MemoryRouter>
  );
  expect(container).toMatchSnapshot();
});
