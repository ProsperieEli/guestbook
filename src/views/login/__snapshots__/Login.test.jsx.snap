// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should login 1`] = `
<div>
  <h1>
    Welcome! Login here!
  </h1>
  <form>
    <label
      for="userName"
    >
      UserName
    </label>
    <input
      id="userName"
      name="userName"
      type="text"
      value=""
    />
    <label
      for="password"
    >
      Password
    </label>
    <input
      id="password"
      name="password"
      type="password"
      value=""
    />
    <button>
      Login
    </button>
  </form>
  <h3 />
</div>
`;
